src/Native/CpuMathNative/CMakeLists.txt

```diff
@@ -2,15 +2,13 @@ project (CpuMathNative)
 
 set(SOURCES
     Sse.cpp
-    Avx.cpp
     MathLinux.S
 )
 
 if(WIN32)
-    set_property(SOURCE Avx.cpp APPEND_STRING PROPERTY COMPILE_FLAGS " /arch:AVX")
+    set_property(SOURCE Sse.cpp APPEND_STRING PROPERTY COMPILE_FLAGS " -mavx")
```

Comment:
Why is this necessary? It wasn't before, was it?

Also, we should follow the typical windows style compile flags: `/arch`.

Reply:
I agree, we should be using `/arch:AVX` for MSVC and `-mavx` for GCC/Clang

Reply:
```
if(WIN32)
    set_property(SOURCE Sse.cpp APPEND_STRING PROPERTY COMPILE_FLAGS " -mavx")
 else()
    set_property(SOURCE Sse.cpp APPEND_STRING PROPERTY COMPILE_FLAGS " -msse3")
    set_property(SOURCE Sse.cpp APPEND_STRING PROPERTY COMPILE_FLAGS " -mavx")
endif
```
is this okay ? just changed the file Avx.cpp to Sse.cpp for all compile flags as Avx.cpp file is no longer present.

Reply:
It is slightly more complicated than that. If we compile using `/arch:AVX` and `-mavx` then it will compile with the VEX encoding, which will only be supported on (most) processors from 2011 or later.

Reply:
so what should this file look like if we remove Avx.cpp as funtions from this file is not used anywhere

Reply:
It works , i was using ```!``` instead of ```NOT``` which was leading to compile error earlier

Reply:
We likely want to be very explicit that ML.NET (on full framework) requires a CPU with SSE3 support (while .NET Core will only require SSE2).

Reply:
so do I need to change something in this file or add some comment here ?

Reply:
No - this now looks fine.

---
In reply to: [226419431](https://github.com/dotnet/machinelearning/pull/1274#discussion_r226419431) [](ancestors = 226419431)